templates/strings/flat-swift4.stencil

```diff
@@ -3,6 +3,7 @@
 
 {% if tables.count > 0 %}
 {% set accessModifier %}{% if param.publicAccess %}public{% else %}internal{% endif %}{% endset %}
+{% set rootTypeName %}{{param.enumName|default:"L10n"}}{% endset %}
```

Comment:
We're on the verge of releasing 6.0 — which will be a breaking version with some changes and with people having to update their config files.

So even if we don't have time to merge this PR in 6.0 (we've already postponed 6.0 for too long so I don't want a PR to block the release if it's not ready by the time we got the release button) and only include it in 6.1, maybe 6.0 could be the occasion to rename the param from `enumName` to `rootTypeName` at least, even before that PR?

@djbe thoughts? You already renamed that param id for ib templates in your PR splitting scene and segue in two separate templates, so we could just do a quick PR only to rename those before 6.0, so that when this PR lands maybe after 6.0, then at least it will not be a breaking change anymore but just an additive one?

Reply:
Oh didn't know about the `Rakefile`. Well `swiftc` should be enough to make sure that the generated source is valid, i.e. it compiles without errors.

What it will not test is that the Obj-C annotations are correct (i.e. are there) and therefore the imports to Obj-C work correctly. So that might be a possible extra test. I don't have any experience with `Rake`, but could try to have a look (if none of you guys would volunteer 😀)

---

Renaming the `enumName` in 6.0 breaking release might be nice addition if there is already breaking release coming.

Reply:
Yeah now that I think about it, compiling the output with `swiftc` should be enough even for ObjC template. After all, we're not generating ObjC code, we're still generating Swift code, just with `@objc` annotations, so 👍 

What could be nice in addition to compiling the generated code to ensure it's valid Swift… would be to compile a small file with code calling that generated Swiftcode (so for example for each strings generated output file we could provide a `.swift` file calling some of the generate constants like `print(L10n.foo)` or such…). And then we could add the same type of calling code to be compile-tested in ObjC as well. That would also be a nice additional test suite to ensure that the call site still looks nice and isn't modified by any modification of our templates, for example. But as of today, we don't have that yet even for Swift — we only compile the generated output, not code calling that generated output. Might be worth a separate, dedicated PR then?

Reply:
Yeah what you are describing is exactly the way I set it up for myself for development and experimenting on feasibility for this PR. I have a small project with one Swift and one Objective-C file printing various localizations using all possible ways (property, method with argument). And the fact it compiles should be good enough verification.